RULES.md

```diff
@@ -1,536 +1,1042 @@
-```java
-/**
- * Validates that every trip in "trips.txt" is used by some stop from "stop_times.txt"
- *
- * <p>Generated notice: {@link UnusedTripNotice}.
- */
-@GtfsValidator
-public class TripUsageValidator extends FileValidator {
-  private final GtfsTripTableContainer tripTable;
-  private final GtfsStopTimeTableContainer stopTimeTable;
-
-  @Inject
-  TripUsageValidator(GtfsTripTableContainer tripTable, GtfsStopTimeTableContainer stopTimeTable) {
-    this.tripTable = tripTable;
-    this.stopTimeTable = stopTimeTable;
-  }
-
-  @Override
-  public void validate(NoticeContainer noticeContainer) {
-    // Do not report the same trip_id multiple times.
-    Set<String> reportedTrips = new HashSet<>();
-    for (GtfsTrip trip : tripTable.getEntities()) {
-      String tripId = trip.tripId();
-      if (reportedTrips.add(tripId) && stopTimeTable.byTripId(tripId).isEmpty()) {
-        noticeContainer.addValidationNotice(new UnusedTripNotice(tripId, trip.csvRowNumber()));
-      }
-    }
-  }
-  /**
-   * A {@code GtfsTrip} should be referred to at least once in {@code GtfsStopTimeTableContainer}
-   * station).
-   *
-   * <p>Severity: {@code SeverityLevel.WARNING}
-   */
-  static class UnusedTripNotice extends ValidationNotice {
-    UnusedTripNotice(String tripId, long csvRowNumber) {
-      super(
-          ImmutableMap.of(
-              "tripId", tripId,
-              "csvRowNumber", csvRowNumber),
-          SeverityLevel.WARNING);
-    }
-  }
-}
-```  
- 
-Note that the notice ID naming conventions changed in `v2` to make contributions of new rules easier by reducing the likelihood of conflicting IDs during parallel development. Please refer to [MIGRATION_V1_V2.md](/docs/MIGRATION_V1_V2.md) for a mapping between v1 and v2 rules.
-
+This document lists all the notices that are emitted by this validator.\
+Note that the notice naming convention changed in `v2` to make contributions of new rules easier by reducing the likelihood of conflicting IDs during parallel development. Please refer to [MIGRATION_V1_V2.md](/docs/MIGRATION_V1_V2.md) for a mapping between v1 and v2 notices.\
+Note that some severities were modified in `v3` to solve discrepancies with the specification. Please refer to [MIGRATION_V2_V3.md](/docs/MIGRATION_V2_V3.md) for a mapping between v2 and v3 notices.
 <a name="definitions"/>
 
 ## Definitions
-Notices are split into three categories: `INFO`, `WARNING`, `ERROR`.
+### A Rule
+A part of the specification that is translated into code in the validator. A Rule will describe if a set of conditions is met or not. For example:
+  - In the specification: in the `stops.txt` file, the field `zone_id` is required if providing fare information using `fare_rules.txt` ([source in the specification](https://gtfs.org/schedule/reference/#stopstxt)).
+  - ```
```

Comment:
Looks like the triple ticks were accidentally added in merging a suggestion - this line can be removed.